libeos-updater-flatpak-installer/perform-flatpak-actions.c

```diff
@@ -69,44 +105,23 @@ try_update_application (FlatpakInstallation       *installation,
 static gboolean
 try_install_application (FlatpakInstallation       *installation,
                          const gchar               *collection_id,
-                         const gchar               *in_remote_name,
+                         const gchar               *remote_name,
                          FlatpakRef                *ref,
                          EosUpdaterInstallerFlags   flags,
                          GError                   **error)
 {
   g_autofree gchar *formatted_ref = flatpak_ref_format_ref (ref);
-  g_autofree gchar *candidate_remote_name = NULL;
-  const gchar *remote_name = in_remote_name;
   gboolean no_pull = !(flags & EU_INSTALLER_FLAGS_ALSO_PULL);
   g_autoptr(GError) local_error = NULL;
 
-  g_assert (in_remote_name != NULL);
-
-  if (collection_id != NULL)
-    {
-      g_message ("Finding remote name for %s", collection_id);
-
-      /* Ignore errors here. We always have the @in_remote_name to use. */
-      candidate_remote_name = euu_lookup_flatpak_remote_for_collection_id (installation,
-                                                                           collection_id,
-                                                                           NULL);
-
-      if (candidate_remote_name != NULL &&
-          g_strcmp0 (in_remote_name, candidate_remote_name) != 0)
-        {
-          g_set_error (error,
-                       EOS_UPDATER_ERROR,
-                       EOS_UPDATER_ERROR_FLATPAK_REMOTE_CONFLICT,
-                       "Specified flatpak remote ‘%s’ conflicts with the remote "
-                       "detected for collection ID ‘%s’ (‘%s’), cannot continue.",
-                       in_remote_name,
-                       collection_id,
-                       candidate_remote_name);
-          return FALSE;
-        }
+  g_assert (remote_name != NULL);
 
-      g_message ("Remote name for %s is %s", collection_id, remote_name);
-    }
+  if (collection_id != NULL &&
+      !validate_collection_id_remote (installation,
```

Comment:
Nitpick: The branch would be a bit clearer if the `validate_collection_id_remote()` function was split out in a preparatory commit. No point doing that now, though, since the review is underway :)